integration_test/tests/power_bi_tests.rs

```diff
@@ -531,4 +531,185 @@ mod integration {
             );
         }
     }
+
+    /// Test PowerBI flow for listing tables
+    ///     - SQLAllocHandle(SQL_HANDLE_STMT)
+    //      - SQLTablesW(null pointer, null pointer, null pointer, "TABLE,VIEW")
+    //      - SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+    //      - SQLGetInfoW(SQL_GETDATA_EXTENSIONS)
+    //      - SQLNumResultCols()
+    //      - For columns 1 to {numCols}
+    //          - SQLColAttributeW(SQL_DESC_CONCISE_TYPE)
+    //          - SQLColAttributeW(SQL_DESC_UNSIGNED)
+    //          - SQLColAttributeW(SQL_COLUMN_NAME)
+    //          - SQLColAttributeW(SQL_COLUMN_NULLABLE)
+    //          - SQLColAttributeW(SQL_DESC_TYPE_NAME)
+    //          - SQLColAttributeW(SQL_COLUMN_LENGTH)
+    //          - SQLColAttributeW(SQL_COLUMN_SCALE)
+    //      - Until SQLFetch returns SQL_NO_DATA
+    //          - SQLFetch()
+    //          - For columns 1 to {numCols}
+    //              - SQLGetData({colIndex}, {defaultCtoSqlType})
+    //      - SQLMoreResults()
+    //      -SQLFreeHandle(SQL_HANDLE_STMT)
+    #[test]
+    fn test_table_listing() {
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = power_bi_connect(env_handle);
+        let mut stmt: Handle = null_mut();
+        let str_len_ptr = &mut 0;
+        let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+            let column_count_ptr = &mut 0;
+            let mut table_view: Vec<u16> = "TABLE,VIEW".encode_utf16().collect();
+            table_view.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLTablesW(
+                    stmt as HStmt,
+                    null_mut(),
+                    0,
+                    null_mut(),
+                    0,
+                    null_mut(),
+                    0,
+                    table_view.as_ptr(),
+                    10
```

Comment:
[nit] Why not using table_view.length - 1 ?

Reply:
Added this 👍